From 72b45af8e8551dd8244ce01f711e9c4a4478db8c Mon Sep 17 00:00:00 2001
From: Kevin Townsend <kevin.townsend@linaro.org>
Date: Wed, 26 Apr 2023 16:33:57 +0000
Subject: [PATCH 03/16] toolchain: gcc: Add __no_instrumentation__ definition

Adds a new '__no_instrumentation__' preprocessor definition for GCC,
mapping it to '__attribute__((__no_instrument_function__))'.

This new macro is used to disable code instrumentation (enabled when
'-finstrument-functions' flag is passed to the compiler) for specific
functions when placed before a function definition.

Signed-off-by: Kevin Townsend <kevin.townsend@linaro.org>
Signed-off-by: Gustavo Romero <gustavo.romero@linaro.org>
---
 include/zephyr/toolchain/gcc.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/zephyr/toolchain/gcc.h b/include/zephyr/toolchain/gcc.h
index 4d3d85858d3..51e02238b3b 100644
--- a/include/zephyr/toolchain/gcc.h
+++ b/include/zephyr/toolchain/gcc.h
@@ -700,6 +700,12 @@ do {                                                                    \
 #define FUNC_NO_STACK_PROTECTOR
 #endif
 
+#if defined(CONFIG_INSTRUMENTATION)
+#define __no_instrumentation__ __attribute__((__no_instrument_function__))
+#else
+#define __no_instrumentation__ /**/
+#endif
+
 #endif /* !_LINKER */
 
 #define TOOLCHAIN_WARNING_ADDRESS_OF_PACKED_MEMBER "-Waddress-of-packed-member"
-- 
2.43.0

